src/wallet/migrate.cpp

```diff
@@ -750,4 +751,18 @@ std::unique_ptr<DatabaseCursor> BerkeleyROBatch::GetNewPrefixCursor(Span<const s
 {
     return std::make_unique<BerkeleyROCursor>(m_database, prefix);
 }
+
+std::unique_ptr<BerkeleyRODatabase> MakeBerkeleyRODatabase(const fs::path& path, const DatabaseOptions& options, DatabaseStatus& status, bilingual_str& error)
+{
+    fs::path data_file = BDBDataFile(path);
+    try {
+        std::unique_ptr<BerkeleyRODatabase> db = std::make_unique<BerkeleyRODatabase>(data_file);
+        status = DatabaseStatus::SUCCESS;
+        return db;
+    } catch (const std::runtime_error& e) {
+        error.original = e.what();
+        status = DatabaseStatus::FAILED_LOAD;
+        return nullptr;
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/26606/commits/e643cac230fc2ba059847bd4e196bddfd97b1e8e ("Add MakeBerkeleyRODatabase"):

Seems like we should have some sort of call to action here prompting the user to open an issue / reach out on IRC if this fails? If we fail to open a bdb database here, the database itself could be fine and it instead might be an issue with our parser.

Reply:
I don't think that's really necessary? I would expect that users would complain if a wallet suddenly was not able to be opened.